WearSpeakerSample/wear/src/main/java/com/example/android/wearable/speaker/ControlDashboard.kt

```diff
@@ -0,0 +1,266 @@
+        targetValue = if (controlDashboardState.micState.expanded) 0.dp else iconCircleRadius,
+        animationSpec = animationSpec
+    )
+
+    val playSize by animateDpAsState(
+        targetValue = if (controlDashboardState.playState.expanded) iconExpandedSize else iconMinimizedSize,
+        animationSpec = animationSpec
+    )
+    val playRadius by animateDpAsState(
+        targetValue = if (controlDashboardState.playState.expanded) 0.dp else iconCircleRadius,
+        animationSpec = animationSpec
+    )
+
+    val musicSize by animateDpAsState(
+        targetValue = if (controlDashboardState.musicState.expanded) iconExpandedSize else iconMinimizedSize,
+        animationSpec = animationSpec
+    )
+    val musicRadius by animateDpAsState(
+        targetValue = if (controlDashboardState.musicState.expanded) 0.dp else iconCircleRadius,
+        animationSpec = animationSpec
+    )
+
+    return ConstraintSet {
+        val circleRef = createRefFor(circle)
+        val micRef = createRefFor(mic)
+        val playRef = createRefFor(play)
+        val musicRef = createRefFor(music)
+
+        constrain(circleRef) { centerTo(parent) }
+        constrain(micRef) {
+            width = Dimension.value(micSize)
+            height = Dimension.value(micSize)
+            circular(circleRef, 0f, micRadius)
+        }
+        constrain(playRef) {
+            width = Dimension.value(playSize)
+            height = Dimension.value(playSize)
+            circular(circleRef, 240f, playRadius)
+        }
+        constrain(musicRef) {
+            width = Dimension.value(musicSize)
+            height = Dimension.value(musicSize)
+            circular(circleRef, 120f, musicRadius)
+        }
+    }
+}
+
+/**
+ * A single control dashboard button
+ */
+@Composable
+private fun ControlDashboardButton(
+    buttonState: ControlDashboardButtonState,
+    transitionInstantly: Boolean,
+    onClick: () -> Unit,
+    layoutId: Any,
+    imageVector: ImageVector,
+    contentDescription: String
+) {
+    val iconPadding = dimensionResource(id = R.dimen.icon_padding)
+    // TODO: Replace with a version of IconButton?
+    //       https://issuetracker.google.com/issues/203123015
+
+    val alpha by animateFloatAsState(
+        targetValue = if (buttonState.visible) 1f else 0f,
+        animationSpec = if (transitionInstantly) snap() else spring(visibilityThreshold = 0.01f)
+    )
+
+    Button(
+        modifier = Modifier
+            .fillMaxSize()
+            .alpha(alpha)
+            .layoutId(layoutId),
+        enabled = buttonState.enabled && buttonState.visible,
+        onClick = onClick
+    ) {
+        Icon(
+            imageVector = imageVector,
+            contentDescription = contentDescription,
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(iconPadding)
+        )
+    }
+}
+
+/**
+ * The state for a single [ControlDashboardButton].
+ */
+data class ControlDashboardButtonState(
```

Comment:
Should this be called `ControlDashboardButtonUiState`? This class is a data class that doesn't handle any UI logic and whose state cannot be mutated. Following our naming conventions, it should be `XUiState`